doc/contributing.md

```diff
@@ -19,14 +12,225 @@ To make a code contribution to the project:
 2. Make the changes you wish to submit on a new branch
 3. Submit a pull request on GitHub.
 
-The other developers will provide feedback, and you may push updates
-into the same branch (which will also update your pull request), until
-general consensus is achieved.
+The other developers will provide feedback, and you may push updates into the same branch (which will also update your pull request), until general consensus is achieved.
 
-For a more detailed explanation, see
-the
-[scikit-image contributors' guide](http://scikit-image.org/docs/stable/contribute.html).
-We follow a very similar process.
+For a more detailed explanation, see the [scikit-image contributors' guide](http://scikit-image.org/docs/stable/contribute.html). We follow a very similar process.
+
+### Adding new features
+
+This section will help you get started adding new features to SkyPortal. We'll walk through adding a new feature: a button that triggers an API call to the back-end, which generates and returns a random string that is stored in the front-end redux state, triggering an automatic re-render of the React component that displays that value in our example component.
```

Comment:
This initial description is a bit too detailed.  Can we start with a very high level description, and explain the details later?